src/borrow_tracker/tree_borrows/mod.rs

```diff
@@ -285,43 +314,76 @@ trait EvalContextPrivExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
 
         let span = this.machine.current_span();
         let alloc_extra = this.get_alloc_extra(alloc_id)?;
-        let range = alloc_range(base_offset, ptr_size);
         let mut tree_borrows = alloc_extra.borrow_tracker_tb().borrow_mut();
+        let prot = new_perm.protector.is_some();
+
+        // Store initialized permissions and their corresponding range.
+        let mut perms_map: RangeMap<LocationState> = RangeMap::new(
+            base_offset + ptr_size,
+            LocationState::new_uninit(
+                Permission::new_disabled(),
+                foreign_access_skipping::IdempotentForeignAccess::None,
+            ),
+        );
```

Comment:
I think this should be initialized with size `ptr_size`, and only cover the place we are retagging. That way we can avoid introducing some `disabled` states here that I don't think we actually want.